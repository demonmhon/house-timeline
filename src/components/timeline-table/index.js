import TimelineTable from './timeline-table';

export default TimelineTable;
