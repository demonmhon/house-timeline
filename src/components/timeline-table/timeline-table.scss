@import '../../scss/variables';

.timeline-container {
  display: flex;
}

.timeline {
  $hour-label-height: 25px;
  $block-width: 30px;
  $block-height: 40px;
  margin: 15px 0 30px;
  padding: 15px 0;

  border: solid 1px $gray;
  border-width: 1px 0;

  h2 {
    color: $font-color;
  }

  &__peoples {
    margin-top: $hour-label-height;
  }

  &__label--hour,
  &__label--people {
    text-align: center;
  }

  &__label--hour {
    width: $block-width * 2;
    height: $hour-label-height;
    font-size: 0.75rem;
    border-right: solid 1px $gray;
  }

  &__label--people {
    height: $block-height;
    padding-right: 20px;
    line-height: $block-height;
    border-right: solid 1px $gray;
  }

  &__hour-block {
    &[data-hour$=':30'] {
      margin-right: -#{$block-width};
      margin-left: calc(-#{$block-width} - 1px);
      .timeline__label--hour {
        visibility: hidden;
      }
    }
  }

  &__people-time-block {
    width: $block-width;
    height: $block-height;
    font-weight: 500;
    font-size: 1.5rem;
    line-height: $block-height;
    text-align: center;
    border: solid 1px $gray;
    cursor: pointer;

    &[data-area='Living Room'] {
      background-color: $living-room;
    }
    &[data-area='Kitchen'] {
      background-color: $kitchen;
    }
    &[data-area='Backyard'] {
      background-color: $backyard;
    }

    & + .timeline__people-time-block {
      border-top: none;
    }
  }

  &__time-table {
    position: relative;
    display: flex;
    padding-bottom: 20px;
    overflow-x: scroll;
  }

  &__area-legends {
    margin: 15px 0;

    @media (min-width: 480px) {
      display: flex;
      justify-content: flex-end;
    }
  }

  &__area-legend {
    position: relative;
    display: block;
    font-size: 0.85rem;

    & + & {
      margin: 15px 0 0;
    }

    &::before {
      display: inline-block;
      width: 25px;
      height: 10px;
      margin-right: 10px;
      content: '';
    }

    @media (min-width: 480px) {
      display: inline-block;

      & + & {
        margin: 0 0 0 20px;
      }
    }

    &[data-area='Living Room']::before {
      background-color: $living-room;
    }
    &[data-area='Kitchen']::before {
      background-color: $kitchen;
    }
    &[data-area='Backyard']::before {
      background-color: $backyard;
    }
  }
}
