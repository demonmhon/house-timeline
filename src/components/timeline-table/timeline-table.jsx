import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';

import { useTimelineContext } from '../../contexts/timeline';
import './timeline-table.scss';

const hourTimeline = [];
const hours = Array(24)
  .fill('')
  .map((a, index) => {
    return index.toString().padStart(2, '0');
  });
hours
  .map((h) => {
    return [`${h}:00`, `${h}:30`];
  })
  .forEach((h) => {
    hourTimeline.push(...h);
  });

const PeopleLabel = (props = {}) => {
  const { name } = props;
  return <div className="timeline__label--people">{name}</div>;
};

const HourLabel = (props = {}) => {
  return <div className="timeline__label--hour">{props.children}</div>;
};

const AreaLagend = (props = {}) => {
  const { name } = props;
  return <div className="timeline__area-legend" data-area={name}>{name}</div>;
};

const PeopleTimeBlock = (props = {}) => {
  const { area } = props;
  return (
    <div
      className="timeline__people-time-block"
      data-tracked={!!area}
      data-area={area}
    ></div>
  );
};

const TimelineTable = (props) => {
  const { timeline, peoples, areas } = useTimelineContext();
  return (
    <div className="timeline">
      <h2>Timeline</h2>
      <div className="timeline-container">
        <div className="timeline__peoples">
          {peoples.map((name) => (
            <PeopleLabel key={name} name={name} />
          ))}
        </div>
        <div className="timeline__time-table">
          {hourTimeline.map((h) => {
            return (
              <div key={h} data-hour={h} className="timeline__hour-block">
                <HourLabel>{h}</HourLabel>
                <div key={h} data-hour={h} className="timeline__used-block">
                  {peoples.map((name) => {
                    return (
                      <PeopleTimeBlock
                        key={`${h}${name}`}
                        area={_.get(timeline, `${name}.${h}`, '')}
                      />
                    );
                  })}
                </div>
              </div>
            );
          })}
        </div>
      </div>
      <div className="timeline__area-legends">
        {areas.map((name) => (
          <AreaLagend key={name} name={name} />
        ))}
      </div>
    </div>
  );
};

TimelineTable.propTypes = {};
TimelineTable.defaultProps = {};

export default TimelineTable;
