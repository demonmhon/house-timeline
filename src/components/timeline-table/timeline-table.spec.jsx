import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { hourTimeline } from '../../contexts/timeline';
import { TimelineTable } from './timeline-table';

describe('<TimelineTable />', () => {
  it('should match snapshot', () => {
    const props = {
      hours: hourTimeline,
      peoples: ['John', 'Dan'],
      areas: ['Bed Room', 'Toilet'],
      timeline: {
        John: {
          '06:00': 'Bed Room',
          '06:30': 'Bed Room',
        },
      },
    };
    const wrapper = shallow(<TimelineTable {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
