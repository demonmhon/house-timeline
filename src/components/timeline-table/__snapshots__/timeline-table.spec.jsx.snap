// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimelineTable /> should match snapshot 1`] = `
<div
  className="timeline"
>
  <h2>
    Timeline
  </h2>
  <div
    className="timeline-container"
  >
    <div
      className="timeline__peoples"
    >
      <PeopleLabel
        key="John"
        name="John"
      />
      <PeopleLabel
        key="Dan"
        name="Dan"
      />
    </div>
    <div
      className="timeline__time-table"
    >
      <div
        className="timeline__hour-block"
        data-hour="00:00"
        key="00:00"
      >
        <HourLabel>
          00:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="00:00"
          key="00:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-00:00"
            name="John"
            onRemove={[Function]}
            time="00:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-00:00"
            name="Dan"
            onRemove={[Function]}
            time="00:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="00:30"
        key="00:30"
      >
        <HourLabel>
          00:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="00:30"
          key="00:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-00:30"
            name="John"
            onRemove={[Function]}
            time="00:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-00:30"
            name="Dan"
            onRemove={[Function]}
            time="00:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="01:00"
        key="01:00"
      >
        <HourLabel>
          01:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="01:00"
          key="01:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-01:00"
            name="John"
            onRemove={[Function]}
            time="01:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-01:00"
            name="Dan"
            onRemove={[Function]}
            time="01:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="01:30"
        key="01:30"
      >
        <HourLabel>
          01:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="01:30"
          key="01:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-01:30"
            name="John"
            onRemove={[Function]}
            time="01:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-01:30"
            name="Dan"
            onRemove={[Function]}
            time="01:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="02:00"
        key="02:00"
      >
        <HourLabel>
          02:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="02:00"
          key="02:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-02:00"
            name="John"
            onRemove={[Function]}
            time="02:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-02:00"
            name="Dan"
            onRemove={[Function]}
            time="02:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="02:30"
        key="02:30"
      >
        <HourLabel>
          02:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="02:30"
          key="02:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-02:30"
            name="John"
            onRemove={[Function]}
            time="02:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-02:30"
            name="Dan"
            onRemove={[Function]}
            time="02:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="03:00"
        key="03:00"
      >
        <HourLabel>
          03:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="03:00"
          key="03:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-03:00"
            name="John"
            onRemove={[Function]}
            time="03:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-03:00"
            name="Dan"
            onRemove={[Function]}
            time="03:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="03:30"
        key="03:30"
      >
        <HourLabel>
          03:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="03:30"
          key="03:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-03:30"
            name="John"
            onRemove={[Function]}
            time="03:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-03:30"
            name="Dan"
            onRemove={[Function]}
            time="03:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="04:00"
        key="04:00"
      >
        <HourLabel>
          04:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="04:00"
          key="04:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-04:00"
            name="John"
            onRemove={[Function]}
            time="04:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-04:00"
            name="Dan"
            onRemove={[Function]}
            time="04:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="04:30"
        key="04:30"
      >
        <HourLabel>
          04:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="04:30"
          key="04:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-04:30"
            name="John"
            onRemove={[Function]}
            time="04:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-04:30"
            name="Dan"
            onRemove={[Function]}
            time="04:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="05:00"
        key="05:00"
      >
        <HourLabel>
          05:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="05:00"
          key="05:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-05:00"
            name="John"
            onRemove={[Function]}
            time="05:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-05:00"
            name="Dan"
            onRemove={[Function]}
            time="05:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="05:30"
        key="05:30"
      >
        <HourLabel>
          05:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="05:30"
          key="05:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-05:30"
            name="John"
            onRemove={[Function]}
            time="05:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-05:30"
            name="Dan"
            onRemove={[Function]}
            time="05:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="06:00"
        key="06:00"
      >
        <HourLabel>
          06:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="06:00"
          key="06:00"
        >
          <PeopleTimeBlock
            area="Bed Room"
            hasOther={false}
            key="John-06:00"
            name="John"
            onRemove={[Function]}
            time="06:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-06:00"
            name="Dan"
            onRemove={[Function]}
            time="06:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="06:30"
        key="06:30"
      >
        <HourLabel>
          06:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="06:30"
          key="06:30"
        >
          <PeopleTimeBlock
            area="Bed Room"
            hasOther={false}
            key="John-06:30"
            name="John"
            onRemove={[Function]}
            time="06:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-06:30"
            name="Dan"
            onRemove={[Function]}
            time="06:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="07:00"
        key="07:00"
      >
        <HourLabel>
          07:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="07:00"
          key="07:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-07:00"
            name="John"
            onRemove={[Function]}
            time="07:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-07:00"
            name="Dan"
            onRemove={[Function]}
            time="07:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="07:30"
        key="07:30"
      >
        <HourLabel>
          07:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="07:30"
          key="07:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-07:30"
            name="John"
            onRemove={[Function]}
            time="07:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-07:30"
            name="Dan"
            onRemove={[Function]}
            time="07:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="08:00"
        key="08:00"
      >
        <HourLabel>
          08:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="08:00"
          key="08:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-08:00"
            name="John"
            onRemove={[Function]}
            time="08:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-08:00"
            name="Dan"
            onRemove={[Function]}
            time="08:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="08:30"
        key="08:30"
      >
        <HourLabel>
          08:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="08:30"
          key="08:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-08:30"
            name="John"
            onRemove={[Function]}
            time="08:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-08:30"
            name="Dan"
            onRemove={[Function]}
            time="08:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="09:00"
        key="09:00"
      >
        <HourLabel>
          09:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="09:00"
          key="09:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-09:00"
            name="John"
            onRemove={[Function]}
            time="09:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-09:00"
            name="Dan"
            onRemove={[Function]}
            time="09:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="09:30"
        key="09:30"
      >
        <HourLabel>
          09:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="09:30"
          key="09:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-09:30"
            name="John"
            onRemove={[Function]}
            time="09:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-09:30"
            name="Dan"
            onRemove={[Function]}
            time="09:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="10:00"
        key="10:00"
      >
        <HourLabel>
          10:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="10:00"
          key="10:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-10:00"
            name="John"
            onRemove={[Function]}
            time="10:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-10:00"
            name="Dan"
            onRemove={[Function]}
            time="10:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="10:30"
        key="10:30"
      >
        <HourLabel>
          10:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="10:30"
          key="10:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-10:30"
            name="John"
            onRemove={[Function]}
            time="10:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-10:30"
            name="Dan"
            onRemove={[Function]}
            time="10:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="11:00"
        key="11:00"
      >
        <HourLabel>
          11:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="11:00"
          key="11:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-11:00"
            name="John"
            onRemove={[Function]}
            time="11:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-11:00"
            name="Dan"
            onRemove={[Function]}
            time="11:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="11:30"
        key="11:30"
      >
        <HourLabel>
          11:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="11:30"
          key="11:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-11:30"
            name="John"
            onRemove={[Function]}
            time="11:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-11:30"
            name="Dan"
            onRemove={[Function]}
            time="11:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="12:00"
        key="12:00"
      >
        <HourLabel>
          12:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="12:00"
          key="12:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-12:00"
            name="John"
            onRemove={[Function]}
            time="12:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-12:00"
            name="Dan"
            onRemove={[Function]}
            time="12:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="12:30"
        key="12:30"
      >
        <HourLabel>
          12:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="12:30"
          key="12:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-12:30"
            name="John"
            onRemove={[Function]}
            time="12:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-12:30"
            name="Dan"
            onRemove={[Function]}
            time="12:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="13:00"
        key="13:00"
      >
        <HourLabel>
          13:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="13:00"
          key="13:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-13:00"
            name="John"
            onRemove={[Function]}
            time="13:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-13:00"
            name="Dan"
            onRemove={[Function]}
            time="13:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="13:30"
        key="13:30"
      >
        <HourLabel>
          13:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="13:30"
          key="13:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-13:30"
            name="John"
            onRemove={[Function]}
            time="13:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-13:30"
            name="Dan"
            onRemove={[Function]}
            time="13:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="14:00"
        key="14:00"
      >
        <HourLabel>
          14:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="14:00"
          key="14:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-14:00"
            name="John"
            onRemove={[Function]}
            time="14:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-14:00"
            name="Dan"
            onRemove={[Function]}
            time="14:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="14:30"
        key="14:30"
      >
        <HourLabel>
          14:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="14:30"
          key="14:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-14:30"
            name="John"
            onRemove={[Function]}
            time="14:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-14:30"
            name="Dan"
            onRemove={[Function]}
            time="14:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="15:00"
        key="15:00"
      >
        <HourLabel>
          15:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="15:00"
          key="15:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-15:00"
            name="John"
            onRemove={[Function]}
            time="15:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-15:00"
            name="Dan"
            onRemove={[Function]}
            time="15:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="15:30"
        key="15:30"
      >
        <HourLabel>
          15:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="15:30"
          key="15:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-15:30"
            name="John"
            onRemove={[Function]}
            time="15:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-15:30"
            name="Dan"
            onRemove={[Function]}
            time="15:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="16:00"
        key="16:00"
      >
        <HourLabel>
          16:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="16:00"
          key="16:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-16:00"
            name="John"
            onRemove={[Function]}
            time="16:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-16:00"
            name="Dan"
            onRemove={[Function]}
            time="16:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="16:30"
        key="16:30"
      >
        <HourLabel>
          16:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="16:30"
          key="16:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-16:30"
            name="John"
            onRemove={[Function]}
            time="16:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-16:30"
            name="Dan"
            onRemove={[Function]}
            time="16:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="17:00"
        key="17:00"
      >
        <HourLabel>
          17:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="17:00"
          key="17:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-17:00"
            name="John"
            onRemove={[Function]}
            time="17:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-17:00"
            name="Dan"
            onRemove={[Function]}
            time="17:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="17:30"
        key="17:30"
      >
        <HourLabel>
          17:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="17:30"
          key="17:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-17:30"
            name="John"
            onRemove={[Function]}
            time="17:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-17:30"
            name="Dan"
            onRemove={[Function]}
            time="17:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="18:00"
        key="18:00"
      >
        <HourLabel>
          18:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="18:00"
          key="18:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-18:00"
            name="John"
            onRemove={[Function]}
            time="18:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-18:00"
            name="Dan"
            onRemove={[Function]}
            time="18:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="18:30"
        key="18:30"
      >
        <HourLabel>
          18:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="18:30"
          key="18:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-18:30"
            name="John"
            onRemove={[Function]}
            time="18:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-18:30"
            name="Dan"
            onRemove={[Function]}
            time="18:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="19:00"
        key="19:00"
      >
        <HourLabel>
          19:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="19:00"
          key="19:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-19:00"
            name="John"
            onRemove={[Function]}
            time="19:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-19:00"
            name="Dan"
            onRemove={[Function]}
            time="19:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="19:30"
        key="19:30"
      >
        <HourLabel>
          19:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="19:30"
          key="19:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-19:30"
            name="John"
            onRemove={[Function]}
            time="19:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-19:30"
            name="Dan"
            onRemove={[Function]}
            time="19:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="20:00"
        key="20:00"
      >
        <HourLabel>
          20:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="20:00"
          key="20:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-20:00"
            name="John"
            onRemove={[Function]}
            time="20:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-20:00"
            name="Dan"
            onRemove={[Function]}
            time="20:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="20:30"
        key="20:30"
      >
        <HourLabel>
          20:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="20:30"
          key="20:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-20:30"
            name="John"
            onRemove={[Function]}
            time="20:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-20:30"
            name="Dan"
            onRemove={[Function]}
            time="20:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="21:00"
        key="21:00"
      >
        <HourLabel>
          21:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="21:00"
          key="21:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-21:00"
            name="John"
            onRemove={[Function]}
            time="21:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-21:00"
            name="Dan"
            onRemove={[Function]}
            time="21:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="21:30"
        key="21:30"
      >
        <HourLabel>
          21:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="21:30"
          key="21:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-21:30"
            name="John"
            onRemove={[Function]}
            time="21:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-21:30"
            name="Dan"
            onRemove={[Function]}
            time="21:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="22:00"
        key="22:00"
      >
        <HourLabel>
          22:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="22:00"
          key="22:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-22:00"
            name="John"
            onRemove={[Function]}
            time="22:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-22:00"
            name="Dan"
            onRemove={[Function]}
            time="22:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="22:30"
        key="22:30"
      >
        <HourLabel>
          22:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="22:30"
          key="22:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-22:30"
            name="John"
            onRemove={[Function]}
            time="22:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-22:30"
            name="Dan"
            onRemove={[Function]}
            time="22:30"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="23:00"
        key="23:00"
      >
        <HourLabel>
          23:00
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="23:00"
          key="23:00"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-23:00"
            name="John"
            onRemove={[Function]}
            time="23:00"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-23:00"
            name="Dan"
            onRemove={[Function]}
            time="23:00"
          />
        </div>
      </div>
      <div
        className="timeline__hour-block"
        data-hour="23:30"
        key="23:30"
      >
        <HourLabel>
          23:30
        </HourLabel>
        <div
          className="timeline__used-block"
          data-hour="23:30"
          key="23:30"
        >
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="John-23:30"
            name="John"
            onRemove={[Function]}
            time="23:30"
          />
          <PeopleTimeBlock
            area=""
            hasOther={false}
            key="Dan-23:30"
            name="Dan"
            onRemove={[Function]}
            time="23:30"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="timeline__area-legends"
  >
    <AreaLagend
      key="Bed Room"
      name="Bed Room"
    />
    <AreaLagend
      key="Toilet"
      name="Toilet"
    />
  </div>
</div>
`;
